playback/src/player.rs

```diff
@@ -1305,14 +1305,20 @@ impl PlayerInternal {
                                 // Engineering Society, 60, 399-408.
                                 if self.config.normalisation_method == NormalisationMethod::Dynamic
                                 {
-                                    // steps 1 + 2: half-wave rectification and conversion into dB
-                                    let abs_sample_db = ratio_to_db(sample.abs());
+                                    if !(self.normalisation_integrator > 0.0
+                                        || self.normalisation_peak > 0.0)
+                                    {
+                                        continue;
+                                    }
 
+                                    let ratio = sample.abs();
                                     // Some tracks have samples that are precisely 0.0, but ratio_to_db(0.0)
                                     // returns -inf and gets the peak detector stuck.
-                                    if !abs_sample_db.is_normal() {
+                                    if ratio == 0.0 {
                                         continue;
                                     }
+                                    // steps 1 + 2: half-wave rectification and conversion into dB
```

Comment:
"Half-wave rectification" is chique wording for doing `sample.abs`. I kept the wording to match the AES article. Please make sure the comment matches that code.

Reply:
I'm familiar with the term "Half-wave rectification" it's used when converting voltage in power supplies. In this case the function is to make sure that "sample_abs" is always a positive number. abs is basically the diode.

Reply:
For a guy who says he's into heavy machinery, you're pretty knowledgeable about analog circuits 👍 must be heavy diodes then 😆

Reply:
> For a guy who says he's into heavy machinery, you're pretty knowledgeable about analog circuits +1 must be heavy diodes then laughing

I am a bucket of useless and disjointed knowledge,lol!!!